### DISCUSSION TITLE
Question on your build environment

### DISCUSSION THREAD

**Discussion by dmshimself** [theengs/gateway#79] [General]
Hi - I thought I'd have a crack and building from source as I have some TPMS issues which I think are all to do with the manufacturer data being published.  Anyway I ran through the build from source option on an Ubuntu 21 system and I get the following, so any advice appreciated on the best build rig.

 pip3 install TheengsGateway-version-tag.tar.gz 
Defaulting to user installation because normal site-packages is not writeable
Processing ./TheengsGateway-version-tag.tar.gz
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: paho-mqtt>=1.6.1 in /home/dms/.local/lib/python3.10/site-packages (from TheengsGateway===version-tag) (1.6.1)
Requirement already satisfied: bleak>=0.15.0 in /home/dms/.local/lib/python3.10/site-packages (from TheengsGateway===version-tag) (0.19.5)
Requirement already satisfied: dbus-fast<2.0.0,>=1.22.0 in /home/dms/.local/lib/python3.10/site-packages (from bleak>=0.15.0->TheengsGateway===version-tag) (1.75.0)
Requirement already satisfied: async-timeout<5,>=3.0.0 in /home/dms/.local/lib/python3.10/site-packages (from bleak>=0.15.0->TheengsGateway===version-tag) (4.0.2)
Building wheels for collected packages: TheengsGateway
  Building wheel for TheengsGateway (pyproject.toml) ... done
  Created wheel for TheengsGateway: filename=TheengsGateway-version_tag-cp310-cp310-linux_x86_64.whl size=311879 sha256=1a7d9d7799bd2b264850ae0a3983ad33983cd5fb81c55cee2f7d3f30ac4abdb3
  Stored in directory: /home/dms/.cache/pip/wheels/de/e8/4a/0cda0b6007e384a10d5a3eb570209f3301f01a20001496d759
  WARNING: Built wheel for TheengsGateway is invalid: Metadata 1.2 mandates PEP 440 version, but 'version-tag' is not
Failed to build TheengsGateway
ERROR: Could not build wheels for TheengsGateway, which is required to install pyproject.toml-based projects

**Comment by koenvervloesem** [comment]
Yeah you need to change the version in setup.py to a valid version string instead of the placeholder version-tag before building the package.

**Reply by koenvervloesem** [reply]
To be more precise, in this line:

https://github.com/theengs/gateway/blob/development/setup.py#L21

change `version_tag` to something like `0.6.0`, then run:

```
python3 setup.py sdist
cd dist
pip3 install TheengsGateway-0.6.0.tar.gz
```

**Comment by dmshimself** [comment]
Many thanks for that prompt advice :-)

**Comment by 1technophile** [comment]
We should add it to the doc maybe, feel free to submit a PR @dmshimself 
https://github.com/theengs/gateway/edit/development/docs/install/install.md

**Comment by dmshimself** [comment]
I'm embarrassed to say that I haven't managed to work out how to pop in 
the right values or even work out what values to use.  I use github a 
fair amount but almost exclusively for cloning bits and pieces for local 
usage in archlinux.  So I'm not the best person to do this bit and in 
fact haven't yet worked out what to change in detail!

**Reply by 1technophile** [reply]
No problem we will do it

**Comment by koenvervloesem** [comment]
I made the steps more explicit in the documentation: https://github.com/theengs/gateway/pull/80/files
